src/Controls/src/Core/Layout/GridLayout.cs

```diff
@@ -1,26 +1,67 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 using Microsoft.Maui.Layouts;
 
 // This is a temporary namespace until we rename everything and move the legacy layouts
 namespace Microsoft.Maui.Controls.Layout2
 {
 	public class GridLayout : Layout, IGridLayout
 	{
-		readonly List<IGridRowDefinition> _rowDefinitions = new();
-		readonly List<IGridColumnDefinition> _columnDefinitions = new();
+		[TypeConverter(typeof(ColumnDefinitionCollectionTypeConverter))]
+		public ColumnDefinitionCollection ColumnDefinitions
+		{
+			get; set;
+		} = new ColumnDefinitionCollection { new ColumnDefinition { Width = GridLength.Auto } };
```

Comment:
Do we need a row and column? Is this unexpected to have when I go add 2 columns to my grid, but now I got 3? I think this is logically/technically correct, but will this be different from the old grid enough that we should rather have an implicit row/col?

Reply:
Yeah, that's a good point. The old Grid has implied rows/columns if you don't add anything (GridCalc just adds them when it starts measuring), which is weird and unexpected. But it does mean that you can just create a Grid, add a child, and without anything else a single row/column is implied. Which is sort of what I was trying to do here, but it _will_ make add/remove confusing. 

Also, this code is defaulting the implied row/column to Auto, and the old Grid used Star for that (which makes more sense). 

I'll take another look at this in the morning; I think we can just have the layout manager add the implied row/column during layout if they're missing.

Reply:
@mattleibow Okay, I've updated this to leave the collections empty upon initialization. So adding new rows/columns shouldn't do anything surprising. And if no rows/columns are added, the layout manager subs in a single implied row/column where appropriate.